src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -315,6 +315,35 @@ void M()
 }");
         }
 
+        [WorkItem(33345, "https://github.com/dotnet/roslyn/issues/33345")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments4()
+        {
+            await TestInRegularAndScriptAsync(
+                @"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+
+        //suffix comment
+        if (x != null)
+        {
+        }
+    }
+}",
+                @"class C
+{
+    void M()
+    {
+        //suffix comment
+        if (o is string x)
+        {
+        }
+    }
+}");
```

Comment:
can we have a test where there are statement before `x = o as string;`?

Reply:
Certainly, see de8a420f011d23b9e4dbab65b1fab00ee758ae38.

As you can see, blank lines between the statement to remove and the next one are removed, but nothing changes above the removed statement.

I guess I'm not exactly sure what the desired behavior is here. I could imagine that we would only want to remove blank lines before a statement if that statement is (or will be, after the removal) the first statement in a block.